### ISSUE TITLE
Status bar on Android switches to not-translucent after mounting StatusBar component

### ISSUE THREAD

**Issue by llamaluvr** [expo/expo#8436]
### Environment

Expo CLI 3.20.9 environment info:
System:
OS: macOS 10.15.4
Shell: 3.2.57 - /bin/bash
Binaries:
Node: 10.17.0 - ~/.nvm/versions/node/v10.17.0/bin/node
Yarn: 1.22.0 - /usr/local/bin/yarn
npm: 6.11.3 - ~/.nvm/versions/node/v10.17.0/bin/npm
Watchman: 4.9.0 - /usr/local/bin/watchman
IDEs:
Android Studio: 3.5 AI-191.8026.42.35.6010548
Xcode: 11.3.1/11C504 - /usr/bin/xcodebuild
npmPackages:
expo: ^37.0.10 => 37.0.10
react: ~16.9.0 => 16.9.0
react-dom: ~16.9.0 => 16.9.0
react-native: ~0.61.5 => 0.61.5
react-native-web: ~0.11.7 => 0.11.7
react-navigation: 4.0.10 => 4.0.10
npmGlobalPackages:
expo-cli: 3.20.9

### Steps to Reproduce

1. Create an app with the default `app.json` settings, which include a translucent Android status bar
2. Add `<StatusBar />` or `<StatusBar translucent />` (or really any `StatusBar` component other than one that explicitly sets `translucent` to false) to render.
3. Run on Android.
4. The status bar will not be translucent as soon as the `StatusBar` is mounted, even if `translucent` is true.

### Expected Behavior

As long as `translucent` is not set to false or `backgroundColor` is not set on the `StatusBar` component, mounting a `StatusBar` component should not remove the default translucency. This is how it behaved on Expo 36 with the Expo 36 client and Expo 36 standalone apps.

### Actual Behavior

Mounting a `StatusBar` component removes the translucency and also shifts the entire view down to accommodate the non-translucent status bar.

This happens on Expo 37 with both the Expo 37 client and standalone apps.
This also happens when running an Expo 36 client on the Expo 37 client, but does not happen in the Expo 36 standalone app.

### Reproducible Demo

Snack demonstrating issue: https://snack.expo.io/@llamaluvr/expo-37-statusbar-issue

This shows a common scenario- hiding a `StatusBar` and then showing it again. Usually this is done in conjunction with react-navigation (e.g., hide `StatusBar` on one screen, show on another). I left that out, but the effect is the same with navigation.

Press "Hide" and then "Show" and the status bar will no longer be translucent.
Also, you can try just pressing "Show" and the same thing happens. Just the act of mounting the `StatusBar` component causes the issue.

Here's a control experiment using the imperative API: https://snack.expo.io/@llamaluvr/201cf5
The imperative API does not have the same issue, though I'm not sure if it's practical to refactor all component uses of `StatusBar` to the imperative API.

**Comment by byCedric** [comment]
Hi @llamaluvr! Thanks for the detailed issue and example! Could it be related to [this unsolved issue in React Native](https://github.com/facebook/react-native/issues/27035)? If so, we might have to fix this upstream first. We will get to this one asap!

**Comment by llamaluvr** [comment]
@byCedric I'm not sure that issue would have caused it alone. This wasn't a problem in Expo 36 and I thought Expo 37 was using the same version of React Native. However, significant changes were made to Expo Android StatusBar behavior as part of Expo 37 (https://github.com/expo/expo/pull/6506), so it's possible the React Native bug presents itself now because of this.

**Comment by llamaluvr** [comment]
I should also add that one other adverse side effect that I observed in the wild but couldn't make a decent snack out. When I upgraded our Managed Standalone Expo app from 36 to 37, the keyboard handling (which uses `KeyboardAvoidingView`) worked fine until mounting a `StatusBar` component, and then after that, the "squishing" of the view to accommodate the keyboard was off by approximately a status bar-sized space (so a text input on a chat might be hidden). I did notice that `Dimensions` reported the same window/ screen height both before and after invoking the `StatusBar` bug, and I was using "height" behavior on `KeyboardAvoidingView`. Anyway, it seems like this sort of behavior is being looked at as a whole as part of https://github.com/expo/expo/issues/7815, but be aware status bar and keyboard avoidance intersect in a myriad of ways!
